From f1418b413f966abb3e277e4c4adb1da59f63106d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian@centricular.com>
Date: Tue, 31 May 2016 14:10:36 +0000
Subject: [PATCH 1/2] qtdemux: Use the demuxer segment instead of a new one for
 MSS streams

Upstream might have told us something about the to be expected segment, so
let's use that information instead of coming up with a [0,-1] segment.

https://bugzilla.gnome.org/show_bug.cgi?id=767071
---
 gst/isomp4/qtdemux.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/gst/isomp4/qtdemux.c b/gst/isomp4/qtdemux.c
index b0fc80b..f05d93e 100644
--- a/gst/isomp4/qtdemux.c
+++ b/gst/isomp4/qtdemux.c
@@ -6203,10 +6203,9 @@ gst_qtdemux_process_adapter (GstQTDemux * demux, gboolean force)
             /* in MSS we need to expose the pads after the first moof as we won't get a moov */
             if (demux->mss_mode && !demux->exposed) {
               if (!demux->pending_newsegment) {
-                GstSegment segment;
-                gst_segment_init (&segment, GST_FORMAT_TIME);
                 GST_DEBUG_OBJECT (demux, "new pending_newsegment");
-                demux->pending_newsegment = gst_event_new_segment (&segment);
+                demux->pending_newsegment =
+                    gst_event_new_segment (&demux->segment);
               }
               qtdemux_expose_streams (demux);
             }
-- 
1.8.3.2

